submissions/AriadnaKravchuk/opp-tiny-js-world/index.js

```diff
@@ -0,0 +1,124 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/AriadnaKravchuk/a-tiny-JS-world
+   Web app: https://ariadnakravchuk.github.io/a-tiny-JS-world
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+
+const SPECIES = {
+   HUMAN: "human",
+   CAT: "cat",
+   DOG: "dog"
+}
+
+const GENDER = {
+   MALE: "male",
+   FEMALE: "female"
+}
+
+const SAYING = {
+   DOG: "woof-woof!",
+   CAT: "meow!"
+}
+
+class Inhabitant {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.friends = [];
+      this.attributes = [
+         this.species,
+         this.name,
+         this.gender,
+         this.saying
+      ];
+   }
+
+   addFriends(...friends) {
+      friends.map((friend) => {
+         if (!this.friends.includes(friend)) this.friends.push(friend);
+      });
+   }
+
+   getFriendsNames() {
+      return this.friends.map(({name}) => name);
+   }
+
+   getAttributes() {
+      return this.attributes;
+   }
+
+   printAttributes() {
+      print(this.getAttributes().join("; ") + "; " + this.getFriendsNames().join(", "));
```

Comment:
Try this and take some learning:
```
[
 this.getAttributes(),
 this.getFriendsNames().join(", "),
].join("; ");
```
You will need to recognize patterns for arrays and `Array` methods often. Worth mastering this kung-fu.
Feel free asking questions.

Also the previous comment is still valid. Need to address this.
> (1) Make class agnostic of environment it works in.
> Let the method just return whatever it needs to create and other code to use it (e.g. print).

Let the method return a string representing an object and use print in line 123.